src/coreclr/jit/morph.cpp

```diff
@@ -10870,6 +10870,97 @@ GenTree* Compiler::fgOptimizeHWIntrinsic(GenTreeHWIntrinsic* node)
             return node;
         }
 
+#if defined(TARGET_XARCH) || defined(TARGET_ARM64)
+
+#if defined(TARGET_XARCH)
+        case NI_SSE_And:
+        case NI_SSE2_And:
+        case NI_AVX_And:
+        case NI_AVX2_And:
+#elif defined(TARGET_ARM64)
+        case NI_AdvSimd_And:
+#endif
+        {
+            assert(node->GetOperandCount() == 2);
+
+            GenTree*            op1      = node->Op(1);
+            GenTree*            op2      = node->Op(2);
+            GenTree*            lhs      = nullptr;
+            GenTree*            rhs      = nullptr;
+            GenTreeHWIntrinsic* inner_hw = nullptr;
+
+            // Transforms ~v1 & v2 to VectorXxx.AndNot(v1, v2)
+            if (op1->OperIs(GT_HWINTRINSIC))
+            {
+                GenTreeHWIntrinsic* xor_hw = op1->AsHWIntrinsic();
+                switch (xor_hw->GetHWIntrinsicId())
+                {
+#if defined(TARGET_XARCH) || defined(TARGET_ARM64)
```

Comment:
This is unnecessary, you're already in a larger identical `ifdef` from L10873.

That being said, the larger identical `ifdef` on L10873 should also be unnecessary given we're in a greater `#ifdef FEATURE_HW_INTRINSICS`